Code/GraphMol/SmilesParse/SmartsWrite.cpp

```diff
@@ -579,6 +585,11 @@ std::string getNonQueryAtomSmarts(const QueryAtom *qatom) {
       res << "+" << qatom->getFormalCharge();
     }
   }
+  int mapNum;
```

Comment:
might want to change this to:

```
if (qatom->getPropIfPresent(common_properties::molAtomMapNumber, mapNum)
     && magnum)
```

To prevent map numbers of 0 from being written out.  I believe the code prevents this, but it could happen.